### ISSUE TITLE
white box artefact produced in SlicePlot

### ISSUE THREAD

**Issue by BenWibking** [yt-project/yt#3889]
### Bug report

**Bug summary**

A white box is inexplicably plotted on top of a SlicePlot.

**Code for reproduction**

The dataset that reproduces this is here: https://cloudstor.aarnet.edu.au/plus/s/eWh5FxbrqaLaj09

```python
import yt
ds = yt.load('plt99000')
p = yt.SlicePlot(ds, "z", ("boxlib", "scalar"), buff_size=(1000,1000))
p.show()
```

**Actual outcome**

![plt99000_Slice_z_scalar](https://user-images.githubusercontent.com/1781919/162130879-f9398049-91a9-4569-9cf1-de69795bc0f1.png)

**Expected outcome**

The data can be plotted correctly using VisIt:
![AMR0035](https://user-images.githubusercontent.com/1781919/162130976-f04bed12-724b-40c7-9636-ec8d669e9c0e.png)

**Version Information**

  * Operating System: Rocky Linux 8.5
  * Python Version: 3.10
  * yt version: main (commit a1bce43eeb6ad6d3dfe15db0acf2971c2053a93a)
  * Other Libraries (if applicable):

Python 3.10 is system-provided.
I installed yt with: `pip install git+https://github.com/yt-project/yt.git`

**Comment by neutrinoceros** [comment]
Hi Ben, thanks for reporting.
This is likely a selection bug, resulting in *no data* being plotted at this particular location, rather than a white box being over plotted.
This is reminiscent of another bug that was recently fixed but only affected particle datasets, while boxlib is grid-only, right ?
Just as a sanity check, can you confirm wether this can be reproduced with yt 4.0.2 ?

**Comment by BenWibking** [comment]
Oh that makes more sense. Boxlib/AMReX does support particles, but my dataset does not have any.

In yt==4.0.2, using the gas density instead (to avoid the vmin/vmax bug in 4.0.2), I get the same issue:
![plt99000_Slice_z_gasDensity](https://user-images.githubusercontent.com/1781919/162145095-ee979d1e-8598-4ac8-9fac-0971e2430f4f.png)

**Comment by neutrinoceros** [comment]
Excellent, thanks for checking. I wouldn't be surprised if this was due to the default slice plane (at z=0) being exactly at the interface between two blocks, resulting in none of them being selected.
Maybe you'd get the correct result by just offsetting the slice plane a bit, e.g.
```python
p = yt.SlicePlot(ds, "z", ("boxlib", "scalar"), center=(0, 0, 1e-6))
```

If my hunch is right, you should also not see this problem with a `ProjectionPlot`.
(I'd test these myself but right now my network is too weak  to get the dataset)

**Comment by neutrinoceros** [comment]
I confirm that
- I can reproduce the problem
- offsetting the center hides the issue
![2](https://user-images.githubusercontent.com/14075922/162165265-a9b40c74-2628-420f-ab21-15fa9f235496.png)


- projections do not suffer from this problem

![3](https://user-images.githubusercontent.com/14075922/162165557-799cc351-fbdc-4568-902e-8fb9d94cc0fc.png)

**Comment by neutrinoceros** [comment]
I confirm that
- I can reproduce the problem
- offsetting the center hides the issue
![2](https://user-images.githubusercontent.com/14075922/162165265-a9b40c74-2628-420f-ab21-15fa9f235496.png)


- projections do not suffer from this problem

![3](https://user-images.githubusercontent.com/14075922/162165557-799cc351-fbdc-4568-902e-8fb9d94cc0fc.png)

**Comment by neutrinoceros** [comment]
So we still need to confirm that we understand why this happens, but if I am correct then I'm not too hopeful that this _can_ be fixed (see https://github.com/yt-project/yt/issues/3828)

**Comment by neutrinoceros** [comment]
So we still need to confirm that we understand why this happens, but if I am correct then I'm not too hopeful that this _can_ be fixed (see https://github.com/yt-project/yt/issues/3828)

**Comment by BenWibking** [comment]
Why can't this be fixed? (For instance, VTK handles this situation correctly.) How is the rendering implemented in yt?

**Comment by neutrinoceros** [comment]
Let me rephrase that: I don't currently know how this could be fixed, because it's supposed to work already. Normally when a slice plane lands right on the edge of two blocks, the left one is selected and the right one isn't. This is where this rule is defined

https://github.com/yt-project/yt/blob/a1bce43eeb6ad6d3dfe15db0acf2971c2053a93a/yt/geometry/_selection_routines/slice_selector.pxi#L100-L102

**Comment by BenWibking** [comment]
Just wanted to check on this. We can workaround by offsetting the slice plane, but it's an obstacle for new users.

Also: I've only noticed this happening for output files that *don't* include ghost cells. Not sure if that gives any clue as to what's going on.

**Comment by chongchonghe** [comment]
@BenWibking The reason I'm asking is because I want to remove the warning here: https://quokka-astro.github.io/quokka/postprocessing/#yt
